src/core/Microsoft.PowerToys.Settings.UI.UnitTests/BackwardsCompatibility/BackCompatTestProperties.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Microsoft.PowerToys.Settings.UI.Lib;
+using Microsoft.PowerToys.Settings.UI.Lib.Interface;
+using Microsoft.PowerToys.Settings.UI.Lib.Utilities;
+using Microsoft.PowerToys.Settings.UI.UnitTests.Mocks;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq.Expressions;
+using System.Text;
+
+namespace Microsoft.PowerToys.Settings.UI.UnitTests.BackwardsCompatibility
+{
+    public static class BackCompatTestProperties
+    {
+        public const string RootPathStubFiles = "..\\..\\..\\..\\src\\core\\Microsoft.PowerToys.Settings.UI.UnitTests\\BackwardsCompatibility\\TestFiles\\{0}\\Microsoft\\PowerToys\\{1}\\{2}";
+
+        internal class MockSettingsRepository<T> : ISettingsRepository<T> where T : ISettingsConfig, new()
+        {
+            T _settingsConfig;
+            ISettingsUtils _settingsUtils;
+            public MockSettingsRepository( ISettingsUtils settingsUtils)
+            {
+                _settingsUtils = settingsUtils;
+            }
```

Comment:
I might be missing something but MockSettingsRepository is not different from Settings Repository and it's also no longer a singleton. Why was this needed?

Reply:
That _is_ the difference.  One of the drawbacks of singletons is that they are difficult to use in unit tests.  There are two reasons this needs to be mocked. 
1) I want to verify that the mocked io provider is read for a particular settings file.  If i use the singleton then then I can't verify the number of times the Read method is called. 
2) While the settings repository is typed, the ISettingsUtils and IOProvider aren't.  This means that the settings repository as a singleton will return the first instance of SettingsUtils that it is created with.    We may want to reconsider the GetInstance implementation in that aspect, but that was oustide the scope of this PR IMHO.

Reply:
Thanks for the clarification. Regarding 2, I agree that we have the issue where SettingsRepository would return the SettingsUtils of the type that it is created with.